src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/RcaController.java

```diff
@@ -225,6 +225,12 @@ public void run() {
         }
         updateRcaState();
 
+        // If RCA is enabled, update Analysis graph with Muted RCAs value
+        if (rcaEnabled) {
+          LOG.debug("Updating Analysis Graph with Muted RCAs");
+          readAndUpdateMutesRcas();
```

Comment:
This call has to be before `updateRcaState` as `updateRcaState` starts the scheduler. The scheduler is started, then it might start its run anytime. Therefore, in theory, it might execute the muted RCA node before coming around to it. Therefore, my suggestion, is move this call before `updateRcaState` this will also require adjusting the call to `rcaConf = RcaControllerHelper.pickRcaConfForRole(currentRole);`